public/app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/QueryLinkExtension.tsx

```diff
@@ -0,0 +1,128 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+
+import { GrafanaTheme2, QueryEditorProps, TimeRange } from '@grafana/data';
+import { getBackendSrv, getPluginLinkExtensions } from '@grafana/runtime';
+import { LinkButton, useStyles2 } from '@grafana/ui';
+
+import { PyroscopeDataSource } from '../datasource';
+import { PyroscopeDataSourceOptions, Query } from '../types';
+
+const EXTENSION_POINT_ID = 'plugins/grafana-pyroscope-datasource/query-links';
```

Comment:
Could this extension point name be more generic? Is there any Pyroscope-specific logic down here? If not maybe this extension point could be leveraged by any App Plugin / any query editor integration. I see there's loading of Pyro's data source settings but I guess that could be a generic type. If you prefer to keep it Pyroscope-specific it'd be probably better to keep in Pyroscope query editor. What do you think?